lib/zendesk_apps_support/validations/manifest.rb

```diff
@@ -63,45 +78,57 @@ def oauth_error(manifest)
         end
 
         def parameters_error(manifest)
-          return unless manifest['parameters']
-
-          unless manifest['parameters'].is_a?(Array)
+          original = manifest.original_parameters
+          unless original.nil? || original.is_a?(Array)
             return ValidationError.new(:parameters_not_an_array)
           end
 
-          para_names = manifest['parameters'].collect { |para| para['name'] }
+          return unless manifest.parameters.any?
+
+          para_names = manifest.parameters.map(&:name)
           duplicate_parameters = para_names.select { |name| para_names.count(name) > 1 }.uniq
           unless duplicate_parameters.empty?
             return ValidationError.new(:duplicate_parameters, duplicate_parameters: duplicate_parameters)
           end
+
+          invalid_required = manifest.parameters.map do |parameter|
+            validate_boolean(parameter.required, "parameters.#{parameter.name}.required")
+          end.compact
+          return invalid_required if invalid_required.any?
+
+          invalid_secure = manifest.parameters.map do |parameter|
+            validate_boolean(parameter.secure, "parameters.#{parameter.name}.secure")
+          end.compact
+          return invalid_secure if invalid_secure.any?
         end
 
         def missing_keys_error(manifest)
-          missing = REQUIRED_MANIFEST_FIELDS.select do |key|
-            manifest[key].nil?
-          end
+          missing = REQUIRED_MANIFEST_FIELDS.map do |ruby_method, manifest_value|
+            manifest_value if manifest.public_send(ruby_method).nil?
+          end.compact
 
           missing_keys_validation_error(missing) if missing.any?
         end
 
         def default_locale_error(manifest, package)
-          default_locale = manifest['defaultLocale']
+          default_locale = manifest.default_locale
           unless default_locale.nil?
             if default_locale !~ Translations::VALID_LOCALE
-              ValidationError.new(:invalid_default_locale, defaultLocale: default_locale)
+              ValidationError.new(:invalid_default_locale, default_locale: default_locale)
             elsif package.translation_files.detect { |file| file.relative_path == "translations/#{default_locale}.json" }.nil?
-              ValidationError.new(:missing_translation_file, defaultLocale: default_locale)
+              ValidationError.new(:missing_translation_file, default_locale: default_locale)
             end
           end
         end
 
         def missing_location_error(package)
-          missing_keys_validation_error(['location']) unless package.has_location?
+          missing_keys_validation_error(['location']) unless package.manifest.location?
         end
 
         def invalid_location_error(package)
           errors = []
-          manifest_locations = package.locations
+          manifest_locations = package.manifest.original_locations
```

Comment:
Feels a little odd to validate `original_locations` after we have already parsed it once.